### PR TITLE
Test coverage for request HTTP2 client entity stream failing

### PR DESCRIPTION
**Pull request by johanandren**
References #3227

### CODE REVIEW

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala [hunk 1]
```diff
@@ -411,6 +412,34 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
...
+    "support stream for request data" should {
+      abstract class StreamingRequestSent extends TestSetup {
+        val streamId = 0x1
+        val requestStream = TestPublisher.probe[ByteString]()
+        user.emitRequest(HttpRequest(
+          protocol = HttpProtocols.`HTTP/2.0`,
+          entity = HttpEntity(ContentTypes.`application/octet-stream`, Source.fromPublisher(requestStream))))
+        network.expectDecodedHEADERS(streamId, endStream = false)
+        requestStream.expectRequest()
+      }
```
Comment (johanandren) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:424]: Kind of duplicating things in #3671 here but the actual test case is not covered by that PR from what I can see.

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala [hunk 2]
```diff
@@ -411,6 +412,34 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
       "fail entity stream if advertised content-length doesn't match" in pending
     }
 
+    "support stream for request data" should {
+      abstract class StreamingRequestSent extends TestSetup {
+        val streamId = 0x1
+        val requestStream = TestPublisher.probe[ByteString]()
+        user.emitRequest(HttpRequest(
+          protocol = HttpProtocols.`HTTP/2.0`,
```
Comment (raboof) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:420]: I don't think setting the protocol is needed

**Review by raboof — APPROVED** [review]

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala (hunk 2 above)
Comment (johanandren) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:420]: Copy pasta from other test where it perhaps makes more sense :) I'll remove.

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala (hunk 1 above)
Comment (jrudolph) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:422]: Would those also be useful for the server side?

**Review by jrudolph — APPROVED** [review]
LGTM

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala (hunk 1 above)
Comment (johanandren) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala:422]: Yes, good point. 

I thought they were but now that I look at it only `send RST_STREAM when entity data stream fails` is there, which is about the response entity only.

**Comment by raboof** [comment]
(needs a rebase)

**Pull request merged**